app/domain/external_invoice/sac_membership/membership_manager.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+class ExternalInvoice::SacMembership::MembershipManager
+  def initialize(person, group, year)
+    @person = person
+    @group = group
+    @year = year
+
+    context = Invoices::SacMemberships::Context.new(Time.zone.today)
+    @member ||= Invoices::SacMemberships::Member.new(person, context)
+  end
+
+  def update_membership_status
+    ActiveRecord::Base.transaction do
+      if stammsektion?
+        extend_membership_duration(@person, @year)
+      elsif neuanmeldung_stammsektion?
+        set_confirmed_at
+        update_role_to_stammsektion_mitglied(@person)
+        @person.household_people.each { |family_member| update_role_to_stammsektion_mitglied(family_member) } if family_main_person?
```

Comment:
das würde ich noch in eine methode extrahieren, damit wir auf der gleichen erzählebene bleiben